tests/serialization_test.py

```diff
@@ -60,6 +61,41 @@ def test_protobuf_serialize_unserialize(self):
         self.assertEqual(sorted(dir(w)),            sorted(dir(unserialized)))
         self.assertEqual(w, unserialized)
 
+    def test_hdf5_serialize_unserialize(self):
+        filename = None
+        try:
+            w = world_gen("Dummy", 32, 16, 1, step=Step.get_by_name("full"))
+            f = tempfile.NamedTemporaryFile(delete=False)
+            f.close()
+            filename = f.name
+            serialized = save_world_to_hdf5(w, filename)
+            unserialized = load_world_to_hdf5(filename)
+            self.assertTrue(_equal(w.elevation['data'], unserialized.elevation['data']))
+            self.assertEqual(w.elevation['thresholds'], unserialized.elevation['thresholds'])
+            self.assertTrue(_equal(w.ocean,             unserialized.ocean))
+            self.assertTrue(_equal(w.biome,             unserialized.biome))
+            self.assertTrue(_equal(w.humidity['quantiles'], unserialized.humidity['quantiles']))
+            self.assertTrue(_equal(w.humidity['data'],  unserialized.humidity['data']))
+            self.assertTrue(_equal(w.humidity,          unserialized.humidity))
+            self.assertTrue(_equal(w.irrigation,        unserialized.irrigation))
+            self.assertTrue(_equal(w.permeability,      unserialized.permeability))
+            self.assertTrue(_equal(w.watermap,          unserialized.watermap))
+            self.assertTrue(_equal(w.precipitation['thresholds'], unserialized.precipitation['thresholds']))
+            self.assertTrue(_equal(w.precipitation['data'], unserialized.precipitation['data']))
+            self.assertTrue(_equal(w.precipitation,     unserialized.precipitation))
+            self.assertTrue(_equal(w.temperature,       unserialized.temperature))
+            self.assertTrue(_equal(w.sea_depth,         unserialized.sea_depth))
+            self.assertEquals(w.seed,                   unserialized.seed)
+            self.assertEquals(w.n_plates,               unserialized.n_plates)
+            self.assertTrue(_equal(w.ocean_level,       unserialized.ocean_level))
+            self.assertTrue(_equal(w.lake_map,          unserialized.lake_map))
+            self.assertTrue(_equal(w.river_map,         unserialized.river_map))
+            self.assertEquals(w.step,                   unserialized.step)
+            self.assertEqual(sorted(dir(w)),            sorted(dir(unserialized)))
```

Comment:
The comparison part should be moved to its own function. I think this piece of code exists three times now. And should another save-format ever be supported, things would get even messier.